### PR TITLE
Retention event query cleanup

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- a part of #5888. refactoring retention so that person prop filtering performance improvements are utilized
- Create a retentioneventsquery class as we have done for funnels, paths, and trends
- Refactor main retention query to use the query class

Followup
- refactor persons queries to use query classes

**Note: there's possibly even more performance improvement opportunity here for refactoring the query to not require a join

### CODE REVIEW

File: posthog/models/filters/retention_filter.py [hunk 1]
```diff
@@ -46,6 +48,8 @@ class RetentionFilter(
     PropertyMixin,
     DisplayDerivedMixin,
     FilterTestAccountsMixin,
+    BreakdownMixin,
+    BreakdownTypeMixin,
```
Comment (EDsCODE) [inline posthog/models/filters/retention_filter.py:52]: Adding unneeded mixins on this filter so it's compatible with the rest of our event query classes. Follow up refactor is adjusting how our filter classes are split

File: ee/clickhouse/queries/retention/retention_event_query.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+from ee.clickhouse.queries.util import get_trunc_func_ch
+from posthog.constants import PAGEVIEW_EVENT, TREND_FILTER_TYPE_ACTIONS, TREND_FILTER_TYPE_EVENTS, TRENDS_LINEAR
+from posthog.models import Entity
+from posthog.models.action import Action
+from posthog.models.filters.retention_filter import RetentionFilter
+
+
+class RetentionEventsQuery(ClickhouseEventQuery):
+    _filter: RetentionFilter
+    _event_query_type: Literal["returning", "target", "target_first_time"]
```
Comment (neilkakkar) [inline ee/clickhouse/queries/retention/retention_event_query.py:14]: Make this an enum? Makes using it in retention class much easier as well, and improves typing 😅

File: ee/clickhouse/queries/retention/retention_event_query.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    _trunc_func: str
+
+    def __init__(self, event_query_type: Literal["returning", "target", "target_first_time"], *args, **kwargs):
+        self._event_query_type = event_query_type
+        super().__init__(*args, **kwargs)
+
+        self._trunc_func = get_trunc_func_ch(self._filter.period)
+
+    def get_query(self) -> Tuple[str, Dict[str, Any]]:
+        _fields = (
```
Comment (neilkakkar) [inline ee/clickhouse/queries/retention/retention_event_query.py:24]: thoughts on following the new format here?:

https://github.com/PostHog/posthog/blob/master/ee/clickhouse/queries/paths/path_event_query.py#L41

I'm yet to update trends, but funnels & paths follow this now. Makes it easier to handle commas + empty strings

**Review by neilkakkar — COMMENTED** [review]
Times like these I'm very happy for tests. Would've been a nightmare to validate otherwise.

**Review by neilkakkar — APPROVED** [review]

**Pull request merged**